includes/api/ApiQueryUsers.php

```diff
@@ -119,18 +120,15 @@ public function execute() {
 		$result = $this->getResult();
 
 		if ( count( $goodNames ) ) {
-			$this->addTables( 'user' );
-			$this->addFields( 'user_id' );
-			$this->addWhereFld( 'user_name', $goodNames );
 
-			// Wikia: there's no point in querying a per-cluster user table for IP blocks
-			#$this->showHiddenUsersAddBlockInfo( isset( $this->prop['blockinfo'] ) );
-
-			$data = array();
-			$res = $this->select( __METHOD__ );
+			$data = [];
+			/* Wikia change begin - SUS-2989 */
+			$dbr = wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
+			$res = $dbr->select('`user`', 'user_id', [ 'user_name' => $goodNames ], __METHOD__);
```

Comment:
Let's add `user_name` column so that `newFromRow` and `getName` does not make a query.